### PR TITLE
Add External Secret Stores support as an alpha feature

### PR DESCRIPTION
**Pull request by turkenh**
### Description of your changes

This PR adds External Secret Stores support as an alpha feature.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

`make reviewable`
`kubectl create -f examples/storeconfig/vault.yaml`
`make run`

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: cmd/provider/main.go [hunk 1]
```diff
@@ -17,26 +17,43 @@ limitations under the License.
...
 
 func main() {
 	var (
 		app            = kingpin.New(filepath.Base(os.Args[0]), "Template support for Crossplane.").DefaultEnvars()
 		debug          = app.Flag("debug", "Run with debug logging.").Short('d').Bool()
-		syncPeriod     = app.Flag("sync", "Controller manager sync period such as 300ms, 1.5h, or 2h45m").Short('s').Default("1h").Duration()
 		leaderElection = app.Flag("leader-election", "Use leader election for the controller manager.").Short('l').Default("false").OverrideDefaultFromEnvar("LEADER_ELECTION").Bool()
+
+		syncInterval     = app.Flag("sync", "How often all resources will be double-checked for drift from the desired state.").Short('s').Default("1h").Duration()
+		pollInterval     = app.Flag("poll", "How often individual resources will be checked for drift from the desired state").Default("1m").Duration()
```
Comment (muvaf) [inline cmd/provider/main.go:52]: Aren't these two the same?

File: internal/controller/template.go [hunk 2]
```diff
@@ -17,23 +17,21 @@ limitations under the License.
 package controller
 
 import (
-	"k8s.io/client-go/util/workqueue"
+	"github.com/crossplane/crossplane-runtime/pkg/controller"
```
Comment (muvaf) [inline internal/controller/template.go:20]: goimport

**Review by muvaf — APPROVED** [review]
Thanks!

File: cmd/provider/main.go (hunk 1 above)
Comment (turkenh) [reply cmd/provider/main.go:52]: Please see https://github.com/crossplane/provider-aws/pull/741#discussion_r660042039

**Pull request merged**